src/components/WalletSelector/AddWalletPending.tsx

```diff
@@ -0,0 +1,152 @@
+        address,
+        signer,
+        fetcher,
+        connector: pendingWallet,
+      });
+      openManageWalletsModal(address);
+
+      return signatureValid;
+    } catch (error: unknown) {
+      if (isWeb3Error(error)) {
+        setDetectedError(error);
+      }
+
+      // Fall back to generic error message
+      if (error instanceof Error) {
+        const web3Error: Web3Error = { code: 'AUTHENTICATION_ERROR', ...error };
+        setDetectedError(web3Error);
+      }
+    }
+  }, [fetcher, openManageWalletsModal, pendingWallet, setDetectedError]);
+
+  const isMetamask = useMemo(() => pendingWalletName.toLowerCase() === 'metamask', [pendingWalletName]);
+
+  const userFriendlyWalletName = useMemo(() => convertWalletName(pendingWalletName), [pendingWalletName]);
+
+  useEffect(() => {
+    async function authenticate() {
+      if (account && signer) {
+        if (authenticatedUserAddresses.includes(account.toLowerCase())) {
+          setPendingState(ADDRESS_ALREADY_CONNECTED);
+          return;
+        }
+
+        if (isMetamask) {
+          setPendingState(CONFIRM_ADDRESS);
+        } else {
+          await attemptAddWallet(account.toLowerCase(), signer);
+        }
+      }
+    }
+
+    void authenticate();
+  }, [account, signer, authenticatedUserAddresses, attemptAddWallet, isMetamask]);
+
+  if (pendingState === ADDRESS_ALREADY_CONNECTED) {
+    return (
+      <div>
+        <StyledTitleMedium>Connect with {userFriendlyWalletName}</StyledTitleMedium>
+        <BodyRegular color={colors.black}>{account?.toLowerCase()}</BodyRegular>
+        <BodyRegular color={colors.gray50}>The provided address is already connected to this account.</BodyRegular>
+        {isMetamask
+        && <BodyRegular color={colors.gray50}>If you want to connect a different address via Metamask, please switch accounts in the extension and try again.</BodyRegular>
+        }
+      </div>);
+  }
+
+  if (pendingState === CONFIRM_ADDRESS && account && signer) {
+    return (
+      <div>
+        <StyledTitleMedium>Connect with {userFriendlyWalletName}</StyledTitleMedium>
+        <BodyRegular color={colors.gray50}>Confirm the following wallet address:</BodyRegular>
+        <BodyRegular color={colors.black}>{account?.toLowerCase()}</BodyRegular>
+        <Spacer height={16}/>
+        <BodyRegular color={colors.gray50}>If you want to connect a different address via Metamask, please switch accounts in the extension and try again.</BodyRegular>
+        <Spacer height={24}/>
+        <StyledButton
+          text="Confirm"
+          onClick={async () => attemptAddWallet(account.toLowerCase(), signer)}
+        />
+      </div>);
+  }
+
+  if (pendingState === PROMPT_SIGNATURE) {
+    return (
+      <div>
+        <StyledTitleMedium>Connect with {userFriendlyWalletName}</StyledTitleMedium>
+        <BodyRegular color={colors.gray50}>Sign the message with your wallet.</BodyRegular>
+      </div>
+    );
+  }
+
+  return (
```

Comment:
add comment above this block to explain this is `pendingState === INITIAL`

Reply:
done ✅